pandas/core/indexes/range.py

```diff
@@ -385,13 +387,13 @@ def tolist(self):
         return list(self._range)
 
     @Appender(Int64Index._shallow_copy.__doc__)
-    def _shallow_copy(self, values=None, **kwargs):
+    def _shallow_copy(self, values=None, name: Label = no_default):
+        name = self.name if name is no_default else name
+
         if values is None:
-            name = kwargs.get("name", self.name)
             return self._simple_new(self._range, name=name)
         else:
-            kwargs.setdefault("name", self.name)
-            return self._int64index._shallow_copy(values, **kwargs)
+            return Int64Index._simple_new(values, name=name)
```

Comment:
why is self._int64index replaced with Int64Index?  does this affect subclassing?

Reply:
Using Int64Index is more direct, avoids constructing self._int64index

Shouldn't affect subclasses, which isn't really supported anyway.